src/adapter/src/session/vars.rs

```diff
@@ -703,8 +700,8 @@ impl SessionVars {
             Ok(&self.emit_timestamp_notice)
         } else if name == EMIT_TRACE_ID_NOTICE.name {
             Ok(&self.emit_trace_id_notice)
-        } else if name == IS_SUPERUSER.name {
-            Ok(&self.is_superuser)
+        } else if name == IS_SUPERUSER_NAME {
+            Ok(&self.user)
```

Comment:
Why not have `Ok(&self.user.is_superuser())` here? That way you don't need to implement `Var` for `User`. It's conceivable that at one point we'd want to add a second session variable that is derived from the `User` object, at which point this wouldn't work.

Reply:
Nope, not possible today in safe code, unfortunately. See e.g. https://github.com/rust-lang/rfcs/issues/3066.